kubeflow/components/generate-dataset.yaml

```diff
@@ -21,18 +25,31 @@ implementation:
     - -c
     - |
       . /opt/genie-toolkit/lib.sh
-      parse_args "$0" "image s3_bucket owner project experiment dataset parallel dataset_path" "$@"
+      parse_args "$0" "image s3_bucket owner project experiment dataset parallel subdatasets target_pruning_size max_turns debug_level s3_datadir" "$@"
       shift $n
-      mkdir workdir 
-      cd workdir
-      aws s3 sync s3://${s3_bucket}/${owner}/workdir/${project}/ .
-      
-      export GENIE_TOKENIZER_ADDRESS=tokenizer.default.svc.cluster.local:8888
+      cd $HOME
+      pwd
+      /opt/genie-toolkit/sync-repos.sh 
+      MAKEDIR=`get_make_dir ${project}`
+      cd workdir/${MAKEDIR}
+
       export TZ=America/Los_Angeles
-      make "-j${parallel}" geniedir=/opt/genie-toolkit thingpedia_cli=thingpedia "experiment=${experiment}" "owner=${owner}" $@ datadir
-      aws s3 sync datadir/ s3://${s3_bucket}/${owner}/dataset/${project}/${experiment}/${dataset}/
-      mkdir -p `dirname $dataset_path`
-      echo "s3://${s3_bucket}/${owner}/dataset/${project}/${experiment}/${dataset}/" > $dataset_path
+      make "-j${parallel}" \
+          geniedir=/opt/genie-toolkit \
+          thingpedia_cli=thingpedia \
+          "experiment=${experiment}" \
+          "owner=${owner}" \
+          "subdatasets=${subdatasets}" \
+          "target_pruning_size=${target_pruning_size}" \
+          "max_turns=${max_turns}" \
+          "debug_level=${debug_level}" \
+          $@ \
+          datadir
+
+      S3_DATADIR=s3://${s3_bucket}/${owner}/dataset/${project}/${experiment}/${dataset}/`date +%s`/
```

Comment:
This is to make run artifacts unique.  So multiple runs will not overwrite other's outputs. It's also useful for data lineage tracking. In case you want to debug and retrain with the dataset from a particular run. Same for model outputs too. If this is not something you want, we can remove it.

Reply:
Yeah I definitely do not want it. It happens often that I try to build a certain dataset multiple times because it crashes, due to bugs. I don't want to then hunt which one actually got uploaded. Besides, it won't work with caching.

Reply:
OK. I've removed the unique/versioned s3 path for both model and dataset.